cmd/podman/volumes/pin.go

```diff
@@ -0,0 +1,66 @@
+package volumes
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containers/podman/v5/cmd/podman/common"
+	"github.com/containers/podman/v5/cmd/podman/registry"
+	"github.com/containers/podman/v5/pkg/domain/entities"
+	"github.com/spf13/cobra"
+)
+
+var (
+	pinDescription = `Mark or unmark a volume as pinned.
+
+Pinned volumes are excluded from system prune operations by default.`
```

Comment:
System reset still catches them, yes?

Reply:
That's a thing we should discuss... My main intention for pinning is to help inexperienced DevOps who are close to reaching capacity limits on their live systems, then execute `prune` or `reset` for some reason, and then realize that the volume containing their live database is gone and the backup is two days old.

After checking the implementation I found that Reset() was NOT respecting the pinned flag. Thanks, I fixed this.

Reply:
I've added an optional solution for `reset`: https://github.com/containers/podman/pull/26846#issuecomment-3367584818